File: 05/TwistyTrampolines.c
#include <stdio.h>
#include <string.h>
#include <sys/timeb.h>

int main(){
    FILE* file = fopen("input.txt", "r");
    int i = 0, len = 0;
    int list[2000];
    int listcopy[2000];

    fscanf(file, "%d", &i);
    while (!feof(file)) {
        list[len] = i;
        ++len;
        fscanf(file, "%d", &i);
    }
    fclose(file);

    memcpy(listcopy, list, 2000 * sizeof(int));

    struct timeb start, end;
    for (int i = 1; i < 3; i++) {
        memcpy(list, listcopy, 2000 * sizeof(int));
        ftime(&start);
        int place = 0;
        int steps = 0;
        while (place >= 0 && place < len) {
            int temp = place;
            place += list[place];
            list[temp] += 1 - 2 * (list[temp] > 2) * (i == 2);
            ++steps;
        }
        ftime(&end);
        float diff = (end.millitm - start.millitm);
        printf("Part %d: %d\tTime: %1.3f\n", i, steps, diff / 1000);
    }
    return 0;
}

File: 05/TwistyTrampolines.py
with open("input.txt") as f:
    offsets = map(int, f.readlines())

preserved = offsets[::]

from datetime import datetime
start = datetime.now()
place = 0
steps = 0
while 0 <= place < len(offsets):
    offsets[place] += 1
    place += offsets[place] - 1
    steps += 1

end = datetime.now()
print "Part 1:", steps, "\t\tTime:", (end-start).total_seconds()
offsets = preserved

start = datetime.now()
place = 0
steps = 0
while 0 <= place < len(offsets):
    temp = place + offsets[place]
    offsets[place] += 1 - 2 * (offsets[place] > 2)
    place = temp
    steps += 1

end = datetime.now()
print "Part 2:", steps, "\tTime:", (end-start).total_seconds()
